sprint3/middlewares/validate.js

```diff
@@ -0,0 +1,129 @@
+export function validateProdCreate(req, res, next) {
+  const { name, price, tags } = req.body;
+
+  if (!name || !price) {
+    return res.status(400).json({ error: "필수 항목 누락" });
+  }
+  if (tags !== undefined) {
+    if (!Array.isArray(tags) || !tags.every((tag) => typeof tag === "string")) {
+      return res.status(400).json({ error: "tags는 문자열 배열이여야 합니다." });
+    }
+  }
+  next();
+}
+
+export function validateProdUpdate(req, res, next) {
+  const { tags } = req.body;
+
+  if (tags !== undefined) {
+    if (!Array.isArray(tags) || !tags.every((tag) => typeof tag === "string")) {
+      return res.status(400).json({ error: "tags는 문자열 배열이여야 합니다." });
+    }
+  }
+  next();
+}
+
+export function validateProdQuery(req, res, next) {
+  const allowedKeys = ["offset", "limit", "name", "description"];
+
+  const queryKeys = Object.keys(req.query);
+
+  for (const key of queryKeys) {
+    if (!allowedKeys.includes(key)) {
+      return res.status(400).json({ error: `잘못된 파라미터: ${key}` });
+    }
+  }
+
+  const { name, description } = req.query;
+
+  if (name !== undefined && (typeof name !== "string" || name.trim() === "")) {
```

Comment:
이 로직들도 모듈화해서 관리하면 좋겠죠?


```suggestion
  if (isNotEmptyString(name)) {
```